compiler/src/dotty/tools/dotc/reporting/messages.scala

```diff
@@ -452,21 +452,9 @@ import transform.SymUtils._
   }
 
   class ObjectMayNotHaveSelfType(mdef: untpd.ModuleDef)(using Context)
-  extends SyntaxMsg(ObjectMayNotHaveSelfTypeID) {
-    def msg = em"""${hl("object")}s must not have a self ${hl("type")}"""
-
-    def explain = {
-      val untpd.ModuleDef(name, tmpl) = mdef
-      val ValDef(_, selfTpt, _) = tmpl.self
-      em"""|${hl("object")}s must not have a self ${hl("type")}:
-           |
-           |Consider these alternative solutions:
-           |  - Create a trait or a class instead of an object
-           |  - Let the object extend a trait containing the self type:
-           |
-           |    object $name extends ${selfTpt.show}"""
-    }
-  }
+  extends SyntaxMsg(ObjectMayNotHaveSelfTypeID):
+    def msg = "objects must not have a self type"
```

Comment:
In a situation like this
```
object Foo { foo => }
```
this message does not fully explain the issue as there's no self type given in code - there's only a self alias. So I would recommend to change the message to
`objects must not have a self alias` and since it's not possible to specify a self type without a self alias (even if it was simply `this` instead of the new name) this would cover both self aliases and self types. For consistency `ObjectMayNotHaveSelfType` should also be renamed